core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -281,44 +247,11 @@ public static void setApplicationName(String applicationName) {
     System.setProperty(USER_TOKEN_PROPERTY_NAME, applicationName);
   }
 
-  /**
-   * Creates a secure socket representing a connection to a Cloud SQL instance.
-   *
-   * @return the newly created Socket.
-   * @throws IOException if error occurs during socket creation.
-   */
-  // TODO(berezv): separate creating socket and performing connection to make it easier to test
-  @VisibleForTesting
-  Socket createSslSocket(ConnectionConfig config) throws IOException, InterruptedException {
-    DefaultConnectionInfoCache connectionInfoCache = getConnectionInfoCache(config);
-
-    try {
-      SSLSocket socket = connectionInfoCache.createSslSocket(this.refreshTimeoutMs);
-
-      // TODO(kvg): Support all socket related options listed here:
-      // https://dev.mysql.com/doc/connector-j/en/connector-j-reference-configuration-properties.html
-      socket.setKeepAlive(true);
-      socket.setTcpNoDelay(true);
-
-      String instanceIp = connectionInfoCache.getPreferredIp(config.getIpTypes(), refreshTimeoutMs);
-
-      socket.connect(new InetSocketAddress(instanceIp, serverProxyPort));
-      socket.startHandshake();
-
-      return socket;
-    } catch (Exception ex) {
-      // TODO(kvg): Let user know about the rate limit
-      connectionInfoCache.forceRefresh();
-      throw ex;
-    }
-  }
-
-  DefaultConnectionInfoCache getConnectionInfoCache(ConnectionConfig config) {
-    return connectionInfoCaches.computeIfAbsent(
-        config.getCloudSqlInstance(), k -> apiFetcher(config));
+  private Connector getConnector(ConnectionConfig config) {
+    return connectors.computeIfAbsent(config.getCloudSqlInstance(), k -> createConnector(config));
```

Comment:
This makes a connector per instance. Instead, we want to use one default connector for all instances unless there are properties that affect its credentials. We'll handle multiple "default connectors" in another PR, yes?

Reply:
That's coming in the next PR.